src/main/java/homework_nr_13/Product.java

```diff
@@ -0,0 +1,63 @@
+package homework_nr_13;
+
+import java.util.ArrayList;
+import java.util.List;
+
+public class Product {
+    private final String name;
+    private final double price;
+    private final Category category;
+
+    public Product(String name, double price, Category category) {
+        this.name = name;
+        this.price = price;
+        this.category = category;
+    }
+
+    public String getName() {
+        return name;
+    }
+
+    public double getPrice() {
+        return price;
+    }
+
+    public Category getCategory() {
+        return category;
+    }
+
+    public String toString(){
+        return name + "(" + category + ") costs " + price;
+    }
+
+    public static void main(String[] args) {
+        List<Product> productList = new ArrayList<>(initPL());
+        System.out.println("---Products cost more than 50:");
+        System.out.println(productList.stream().filter(e -> e.price > 50).map(e -> e.name).toList());
+
+        System.out.print("\nThe most expensive product is ");
+        System.out.println(productList.stream().filter(e -> e.getCategory() == Category.ELECTRONICS)
+                .reduce((Product o1, Product o2) -> o1.getPrice() > o2.getPrice() ? o1 : o2)
```

Comment:
why not use min() here?

Reply:
I wanted to use .max(), but didn't understand how to create comparator.